ocaml-lsp-server/src/compl.ml

```diff
@@ -282,8 +282,15 @@ let complete (state : State.t)
           let prefix =
             prefix_of_position ~short_path:false (Document.source doc) position
           in
+          let deprecated =
+            Option.value
+              ~default:false
+              (let open Option.O in
+              let* item = completion_item_capability in
+              item.deprecatedSupport)
+          in
           if not (Typed_hole.can_be_hole prefix) then
-            Complete_by_prefix.complete merlin prefix pos ~resolve
+            Complete_by_prefix.complete merlin prefix pos ~resolve ~deprecated
```

Comment:
I'm growing to believe that we should have a new type (record) that would contain all completion-related capabilities that is lazily evaluated and passed as a single argument to functions rather than several somewhat confusing named arguments

Reply:
Sure, I don't mind that. I'm not sure I'll get around to doing it soon though. Why does it matter if it's lazy or not?

Reply:
I'll work on that when I have time then. Lazy - just to not to recompute the bool value from the option on every completion if that makes sense.